elastic/kibana
# [Entity Analytics] Asset Criticality soft delete

## Summary

This PR introduces a "soft delete" for Asset Criticality records. Instead of deleting the document, we now simply change the criticality field to the value `"deleted"`.
This value is fully managed on Kibana and should not be exposed to the client side.


### How to test

1. Add some host/user data
2. Make sure to enable the `entityStoreEnabled` feature flag
3. Assign asset criticality to a user/host.
    * You may need to enable Asset Criticality in Kibana Advanced Settings 
5. Verify the criticality has been assigned by `GET .asset-citicality*`.
6. Unassign the criticality for that host/user via the UI.
7. `GET`ing the criticality index should now still show the updated record with `"deleted"` criticality value
8. The Ui should also show criticality as `Unassigned` for the deleted record.



Implements https://github.com/elastic/security-team/issues/10531, which is part of the overall [Entity Store for 8.16](https://github.com/elastic/security-team/issues/10529) work
